PWGEM/PhotonMeson/Tasks/gammaConversions.cxx

```diff
@@ -494,16 +545,52 @@ struct GammaConversions {
     }
   }
 
-  template <typename TV0>
-  void fillV0Histograms(mapStringHistPtr& theContainer, TV0 const& theV0, float const& theV0CosinePA)
+  //this is really ugly but it works. Would be easier if .end() worked..
+  template <typename TTRACKS>
+  void fillhpeDivpGammaHistograms(mapStringHistPtr& theContainer, TTRACKS const& theTwoV0Daughters)
+  {
+    float p_electron[2];
+    Int_t counter = 0;
+
+    for (auto& lTrack : theTwoV0Daughters) {
+      p_electron[counter] = lTrack.p();
+      counter++;
+    }
+
+    fillTH1(theContainer, "hpeDivpGamma", p_electron[0] / (p_electron[0] + p_electron[1]));
+    fillTH1(theContainer, "hpeDivpGamma", p_electron[1] / (p_electron[0] + p_electron[1]));
+  }
+
+  template <typename TV0, typename TTRACKS>
+  void fillV0Histograms(mapStringHistPtr& theContainer, TV0 const& theV0, TTRACKS const& theTwoV0Daughters, float const& theV0CosinePA)
   {
     fillTH1(theContainer, "hEta", theV0.eta());
     fillTH1(theContainer, "hPhi", theV0.phi());
     fillTH1(theContainer, "hPt", theV0.pt());
     fillTH1(theContainer, "hConvPointR", theV0.v0radius());
+    fillTH1(theContainer, "hConvPointZ", theV0.z());
     fillTH1(theContainer, "hCosPAngle", theV0CosinePA);
     fillTH2(theContainer, "hArmenteros", theV0.alpha(), theV0.qtarm());
     fillTH2(theContainer, "hPsiPt", theV0.psipair(), theV0.pt());
+    fillTH2(theContainer, "hRVsZ", theV0.recalculatedVtxR(), theV0.z()); // as long as z recalculation is not fixed use this
+
+    float p_electron[2];
+    Int_t counter = 0;
+
+    // Fill momentum distribution histogram
+    for (auto& lTrack : theTwoV0Daughters) {
+      p_electron[counter] = lTrack.p();
+      counter++;
+    }
+    fillTH1(theContainer, "hpeDivpGamma", p_electron[0] / theV0.pt());
+    fillTH1(theContainer, "hpeDivpGamma", p_electron[1] / theV0.pt());
+  }
+
+  template <typename TV0>
```

Comment:
maybe you can add a comment here, that you define this function in addition to fillV0Histograms because these histos only exist for rec and val